pkg/apis/kubermatic/v1/external_cluster.go

```diff
@@ -43,7 +43,30 @@ type ExternalCluster struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	Spec ExternalClusterSpec `json:"spec,omitempty"`
+	Spec   ExternalClusterSpec   `json:"spec"`
+	Status ExternalClusterStatus `json:"status,omitempty"`
+}
+
+// ExternalClusterStatus denotes status information about an ExternalCluster.
+type ExternalClusterStatus struct {
+	// Conditions contains conditions an externalcluster is in, its primary use case is status signaling for controller
+	Condition ExternalClusterCondition `json:"conditions,omitempty"`
+}
+
+type ExternalClusterCondition struct {
+	Phase ExternalClusterPhase `json:"status"`
```

Comment:
Please ensure the JSON tag matches the field name.

Reply:
oops, i will correct it